platform/security/sts/security-sts-upbstvalidator/src/main/java/org/codice/ddf/security/validator/uname/UPBSTValidator.java

```diff
@@ -317,36 +287,19 @@ public TokenValidatorResponse validateToken(TokenValidatorParameters tokenParame
             Principal principal = createPrincipal(ut.getName(), ut.getPassword(),
                     ut.getPasswordType(), ut.getNonce(), ut.getCreated());
 
-            // Get the realm of the UsernameToken
-            String tokenRealm = null;
-            if (usernameTokenRealmCodec != null) {
-                tokenRealm = usernameTokenRealmCodec.getRealmFromToken(ut);
-                // verify the realm against the cached token
-                if (secToken != null) {
-                    Properties props = secToken.getProperties();
-                    if (props != null) {
-                        String cachedRealm = props.getProperty(STSConstants.TOKEN_REALM);
-                        if (!tokenRealm.equals(cachedRealm)) {
-                            validateTarget.setState(STATE.INVALID);
-                            return response;
-                        }
-                    }
-                }
-            }
-
             // Store the successfully validated token in the cache
             if (tokenParameters.getTokenStore() != null && secToken == null && STATE.VALID
                     .equals(validateTarget.getState())) {
                 secToken = new SecurityToken(ut.getID());
                 secToken.setToken(ut.getElement());
-                int hashCode = ut.hashCode();
+                int hashCode = tokenId.hashCode();
                 String identifier = Integer.toString(hashCode);
                 secToken.setTokenHash(hashCode);
                 tokenParameters.getTokenStore().add(identifier, secToken);
             }
 
             response.setPrincipal(principal);
-            response.setTokenRealm(tokenRealm);
+            response.setTokenRealm(null);
```

Comment:
Since we have the realm from `usernameToken.getRealm()`, we should probably set it.